### PR TITLE
[Backport 2.x] [Workspace] Delete saved objects by workspace

### PR DESCRIPTION
**Pull request by Hailong-am**
backport 735424bcaabfc752453c4d514ddd5d4b4f1b6d6d from #6013

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 - [Multiple Datasource] Add TLS configuration for multiple data sources ([#6171](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6171))
 
+- [Workspace] Add delete saved objects by workspace functionality([#6013](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6013))
+
```
Comment (SuZhou-Joe) [inline CHANGELOG.md:21]:
```suggestion
```
Seems CHANGELOG.md should not have change within backport PR. Release owner will help to collect all the changes inside that release and batch update CHANLOG. 

Better to remove this CHANGE.

**Review by SuZhou-Joe — APPROVED** [review]
LGTM, please make sure all the flows pass.

**Comment by Hailong-am** [comment]
@ZilongX could you help to review the backport PR to 2.x?

**Review by BionIT — APPROVED** [review]

**Comment by BionIT** [comment]
code coverage failed, and can we check?

**Comment by Hailong-am** [comment]
> code coverage failed, and can we check? @Hailong-am

by looking at the coverage report https://app.codecov.io/gh/opensearch-project/OpenSearch-Dashboards/pull/6424/blob/src/core/server/saved_objects/service/lib/repository.ts, it looks like whole file are more covered, And we do have [unit test](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6424/files#diff-8a58e602e79e19512cd392387f38a79ba9782f60d0fc1596a90d5e9e18efb6bd) to cover the change.

It might be something wrong with collecting coverage from test run.

**Pull request merged**